x-pack/plugins/observability/public/components/custom_threshold/components/alert_details_app_section/alert_details_app_section.tsx

```diff
@@ -85,6 +87,36 @@ export default function AlertDetailsAppSection({
   const alertEnd = alert.fields[ALERT_END] ? moment(alert.fields[ALERT_END]).valueOf() : undefined;
   const groups = alert.fields[ALERT_GROUP];
   const tags = alert.fields[TAGS];
+  const CHART_TITLE_LIMIT = 120;
+
+  const generateChartTitle = (criterion: MetricExpression) => {
+    let equationTitle = '';
+    const symbolResolver: Record<string, string> = {};
+    const equation = criterion.equation
+      ? criterion.equation
+      : criterion.metrics.map((m) => m.name).join(' + ');
+
+    criterion.metrics.forEach(
+      (metric) =>
+        (symbolResolver[metric.name] = `${metric.aggType} (${
+          metric.field ? metric.field : metric.filter ? metric.filter : 'all documents'
+        })`)
+    );
+
+    [...equation].forEach((symbol) =>
```

Comment:
I saw that we can pass any name for the aggregation (when creating a rule through API), does it mean if we have multiple letters for aggregation, then this logic will not work properly?

Reply:
Good catch @maryam-saeidi. I have updated the logic in edca4c626a79f713e53ac714fedfffa97cac8b48 to consider having multiple letters in name for aggregation.